app/startup.py

```diff
@@ -44,34 +43,47 @@
 def start_app():
     middlewares = [
         Middleware(ProxyHeadersMiddleware, trusted_hosts=Settings.service_domain),
-        Middleware(
-            TraceRequestMiddleware,
-            sampler=AlwaysOnSampler(),
-            instrumentation_key=Settings.instrumentation_key,
-            cloud_role_name=add_cloud_role_name,
-            extra_attrs=dict(
-                environment=Settings.ENVIRONMENT,
-                server_location=Settings.server_location
-            ),
-            logging_instances=logging_instances
-        )
     ]
 
+    if Settings.ENVIRONMENT != "DEVELOPMENT":
+        from app.middleware.tracers.starlette import TraceRequestMiddleware
+
+        middlewares.append(
+            Middleware(
+                TraceRequestMiddleware,
+                sampler=AlwaysOnSampler(),
+                instrumentation_key=Settings.instrumentation_key,
+                cloud_role_name=add_cloud_role_name,
+                extra_attrs=dict(
+                    environment=Settings.ENVIRONMENT,
+                    server_location=Settings.server_location
+                ),
+                logging_instances=logging_instances
+            )
+        )
+
     if Settings.DEBUG:
         handler = logging.StreamHandler(stdout)
 
         for log, level in logging_instances:
             log.addHandler(handler)
             log.setLevel(level)
 
+    # todo: below is for production the one after that is for development
+
     app = FastAPI(
         title="UK Coronavirus Dashboard - API Service",
         version="2.1.0",
-        docs_url=None,
+        docs_url=(
```

Comment:
Use `Settings.DEBUG` here. Environment name is not a determinant of the development environment, it refers to the DevOps environment.